### ISSUE TITLE
Attempting to install gtk+3 fails on atk dependency

### ISSUE THREAD

**Issue by mcampos-quinn** [Linuxbrew/homebrew-core#7175]
Hi! 

Trying to `brew install gtk+3` fails with the error message `No package 'atk' found`. I can confirm that `atk` is in fact installed. Brew doctor yields `ready to brew`.

I came across this attempting to `brew install amiaopensource/amiaos/ltopers` but tracked it down to gtk+3 and atk. Attempting to `brew install gtk+3` generates the problem.

Here's the requested gist: https://gist.github.com/mcampos-quinn/a748300c0a833f81278a2b4c428886fb

I see that there is an open pull request #6602 to create a linuxbrew bottle for gtk+3 and maybe this issue would be solved with that?

Thanks!

**Comment by arkanoid87** [comment]
Same problem with gtk+ formula

**Comment by iMichka** [comment]
What does this return for you?:
> pkg-config --libs atk

**Comment by arkanoid87** [comment]
$ pkg-config --libs atk
Package atk was not found in the pkg-config search path.
Perhaps you should add the directory containing `atk.pc'
to the PKG_CONFIG_PATH environment variable
No package 'atk' found

2018-05-05 12:56 GMT+02:00 Michka Popoff <notifications@github.com>:

**Comment by iMichka** [comment]
Is brew in your path? Not sure the right pkg-config is being used here.
Can you try like this?
> /home/linuxbrew/.linuxbrew/bin/pkg-config --libs atk

Also please provide your `brew config`. Thanks.

**Comment by arkanoid87** [comment]
```
$ which brew
~/.linuxbrew/bin/brew
```

```
$ which pkg-config
~/.linuxbrew/bin/pkg-config
```

```
$ $HOME/.linuxbrew/bin/pkg-config --libs atk
Package atk was not found in the pkg-config search path.
Perhaps you should add the directory containing `atk.pc'
to the PKG_CONFIG_PATH environment variable
No package 'atk' found
```

```
$ brew config
HOMEBREW_VERSION: >=1.4.0 (shallow or no git repository)
ORIGIN: https://github.com/Linuxbrew/brew
HEAD: d9631dc889be0ce9731e3c07238fb1c2a1a06b4d
Last commit: 18 hours ago
Core tap ORIGIN: https://github.com/Linuxbrew/homebrew-core
Core tap HEAD: 10e4c1ccb5b2278617d2ffce5feb0425d7a7ba52
Core tap last commit: 12 minutes ago
HOMEBREW_PREFIX: /home/nfs/jack/.linuxbrew
HOMEBREW_REPOSITORY: /home/nfs/jack/.linuxbrew
HOMEBREW_CELLAR: /home/nfs/jack/.linuxbrew/Cellar
HOMEBREW_CACHE: /home/nfs/jack/.cache/Homebrew
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_GITHUB_API_TOKEN: set
CPU: 20-core 64-bit haswell
Homebrew Ruby: 2.3.3 => /home/nfs/jack/.linuxbrew/Library/Homebrew/vendor/portable-ruby/2.3.3_2/bin/ruby
Clang: N/A
Git: 2.17.0 => /home/nfs/jack/.linuxbrew/bin/git
Curl: 7.29.0 => /usr/bin/curl
Java: N/A
Kernel: Linux 3.10.0-327.28.3.el7.x86_64 x86_64 GNU/Linux
OS: CentOS Linux release 7.2.1511 (Core)
Host glibc: 2.17
/usr/bin/gcc: 4.8.5
glibc: 2.23
gcc: 5.5.0_4
xorg: 20170115_1
```

**Comment by iMichka** [comment]
`brew list atk`?

This should be there: `lib/pkgconfig/atk.pc`

**Comment by arkanoid87** [comment]
```
$ brew list atk
/home/nfs/jack/.linuxbrew/Cellar/atk/2.28.1_2/include/atk-1.0/ (34 files)
/home/nfs/jack/.linuxbrew/Cellar/atk/2.28.1_2/lib64/girepository-1.0/Atk-1.0.typelib
/home/nfs/jack/.linuxbrew/Cellar/atk/2.28.1_2/lib64/pkgconfig/atk.pc
/home/nfs/jack/.linuxbrew/Cellar/atk/2.28.1_2/lib64/ (3 files)
/home/nfs/jack/.linuxbrew/Cellar/atk/2.28.1_2/share/gir-1.0/Atk-1.0.gir
/home/nfs/jack/.linuxbrew/Cellar/atk/2.28.1_2/share/locale/ (99 files)
```

**Comment by iMichka** [comment]
Maybe this is because it is in a `lib64` folder? I have:
```
/home/linuxbrew/.linuxbrew/Cellar/atk/2.28.1_2/include/atk-1.0/ (34 files)
/home/linuxbrew/.linuxbrew/Cellar/atk/2.28.1_2/lib/girepository-1.0/Atk-1.0.typelib
/home/linuxbrew/.linuxbrew/Cellar/atk/2.28.1_2/lib/pkgconfig/atk.pc
/home/linuxbrew/.linuxbrew/Cellar/atk/2.28.1_2/lib/ (3 files)
/home/linuxbrew/.linuxbrew/Cellar/atk/2.28.1_2/share/gir-1.0/Atk-1.0.gir
/home/linuxbrew/.linuxbrew/Cellar/atk/2.28.1_2/share/locale/ (99 files)
```

Can you try:
`brew uninstall atk && brew install atk`?

**Comment by arkanoid87** [comment]
```
$ HOMEBREW_NO_AUTO_UPDATE=1 brew install gtk+
==> Downloading https://download.gnome.org/sources/gtk+/2.24/gtk+-2.24.32.tar.xz
Already downloaded: /home/nfs/jack/.cache/Homebrew/gtk+-2.24.32.tar.xz
==> Downloading https://bug757187.bugzilla-attachments.gnome.org/attachment.cgi?id=331173
Already downloaded: /home/nfs/jack/.cache/Homebrew/gtk+--patch-ce5adf1a019ac7ed2a999efb65cfadeae50f5de8663638c7f765f8764aa7d931.cgi
==> Patching
==> Applying attachment.cgi
patching file gdk/quartz/gdkimage-quartz.c
==> ./configure --disable-silent-rules --prefix=/home/nfs/jack/.linuxbrew/Cellar/gtk+/2.24.32_2 --disable-glibtest --enable-introspection=yes --with-gdktarget=x11 --disable-visibility --disable
Last 15 lines from /home/nfs/jack/.cache/Homebrew/Logs/gtk+/01.configure:
checking for library containing strerror... none required
checking whether make sets $(MAKE)... (cached) yes
checking for pkg-config... /home/nfs/jack/.linuxbrew/opt/pkg-config/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for BASE_DEPENDENCIES... no
configure: error: Package requirements (glib-2.0 >= 2.28.0    atk >= 1.29.2    pango >= 1.20    cairo >= 1.6    gdk-pixbuf-2.0 >= 2.21.0) were not met:

No package 'atk' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables BASE_DEPENDENCIES_CFLAGS
and BASE_DEPENDENCIES_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

READ THIS: https://github.com/Linuxbrew/brew/wiki/troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core, which support macOS only.

These open issues may also help:
gtk-doc: Build a bottle for Linuxbrew https://github.com/Linuxbrew/homebrew-core/pull/6929
Attempting to install gtk+3 fails on atk dependency https://github.com/Linuxbrew/homebrew-core/issues/7175
No package 'cairo-xlib' found in compiling gtk+(yocto) https://github.com/Linuxbrew/homebrew-core/issues/7199
[jack@geo1 Library]$
```

**Comment by JuPlutonic** [comment]
pkg-config --libs atk
```Package atk was not found in the pkg-config search path.
Perhaps you should add the directory containing `atk.pc'
to the PKG_CONFIG_PATH environment variable
No package 'atk' found
```

**Comment by JuPlutonic** [comment]
Added to L97
```
HOMEBREW_PREFIX/"lib64/pkgconfig",
```
Created directory
```
mkdir -p `brew --repository`/lib64/pkgconfig
```
creatred symlink ln -s $(brew --repository)/Cellar/atk/2.28.1_2/lib/x86_64-linux-gnu/pkgconfig/atk.pc $(brew 00repository/lib64/pkgconfig/atk.pc
then run
HOMEBREW_NO_AUTO_UPDATE=1 brew install gtk+
and got same error with *atk*

**Comment by iMichka** [comment]
@JuPlutonic I see that you are using Ubuntu 18.04. I do too, and I can not reproduce the error.

I think I have another idea. I was looking on how the pkg-config formula is working (https://github.com/Linuxbrew/homebrew-core/blob/master/Formula/pkg-config.rb#L19-L27).
We setup a path and tell pkg-config where to look for .pc files.

Can you execute the following command:
`pkg-config pkg-config --variable pc_path`

This will tell us where pkg-config is looking first. In my case, the first location is:
`/home/linuxbrew/.linuxbrew/lib/pkgconfig`
In there, you can find all the .pc files, which are symlinks to the original .pc file in the Cellar.

Can you check the first path is pointing to the right lib folder? Is the atk.pc file in there? Does it point to the right atk.pc file in Cellar?

I also see this `x86_64-linux-gnu` subfolder in `lib/x86_64-linux-gnu/pkgconfig/atk.pc`, and not sure from where it comes from? I tried to rebuild `pkg-config` and `atk` from source to see if that folder pops up, but I do not see that happen, so I am wondering why it exists at all, and if this is why everything is messed up for you.

**Comment by JuPlutonic** [comment]
1st command gave /opt/.linuxbrew/lib/pkgconfig:/opt/.linuxbrew/share/pkgconfig:/opt/.linuxbrew/Library/Homebrew/os/linux/pkgconfig
/opt/.linuxbrew is mine repository
/opt/.linuxbrew/lib/pkgconfig don't contain atk.pc symlink (but brew linkage --test atk didn't report about that).
/opt/.linuxbrew/share/pkgconfig contains only 8 symlinks
and path /opt/.linuxbrew/Library/Homebrew/os/linux/pkgconfig doesn't exist.

**Comment by JuPlutonic** [comment]
/opt/.linuxbrew/lib/x86_64-linux-gnu (is symlink) -> ../Cellar/atk/2.28.1_2/lib/x86_64-linux-gnu is where actually pkgconfig/atk.pc located.

**Comment by JuPlutonic** [comment]
/opt/.linuxbrew/lib/x86_64-linux-gnu (is symlink) -> ../Cellar/atk/2.28.1_2/lib/x86_64-linux-gnu is where actually pkgconfig/atk.pc located.

**Comment by arkanoid87** [comment]
Here's my pc_path. It is not containing atk.pc
```
[jack@geo1 .linuxbrew]$ pkg-config pkg-config --variable pc_path
/home/nfs/jack/.linuxbrew/lib/pkgconfig:/home/nfs/jack/.linuxbrew/share/pkgconfig:/home/nfs/jack/.linuxbrew/Library/Homebrew/os/linux/pkgconfig
[jack@geo1 .linuxbrew]$ find . -name atk.pc
./Cellar/atk/2.28.1_2/lib64/pkgconfig/atk.pc
```

**Comment by sjackman** [comment]
I think `#{HOMEBREW_LIBRARY}/Homebrew/os/linux/pkgconfig` exists simply for parallelism with Brew on macOS, but this directoy has never existed. I'm fine to remove it.

```
Cellar/atk/2.28.1_2/lib/x86_64-linux-gnu/pkgconfig/atk.pc
Cellar/atk/2.28.1_2/lib64/pkgconfig/atk.pc
```

Ideally this file would be moved to `Cellar/atk/2.28.1_2/lib/pkgconfig/atk.pc`. As a workaround, I'm okay with adding `Cellar/atk/2.28.1_2/lib/x86_64-linux-gnu/pkgconfig` to the search path.

**Comment by arkanoid87** [comment]
in my case `#{HOMEBREW_LIBRARY}/lib/pkgconfig` exists and contains many symlinks to pc files, but not atk.pc

**Comment by arkanoid87** [comment]
I've tried to create manually atk.pc link
```
$ pwd
/home/nfs/jack/.linuxbrew/lib/pkgconfig
$ ls -lah atk.pc
lrwxrwxrwx. 1 jack jack 48 May  6 21:54 atk.pc -> ../../Cellar/atk/2.28.1_2/lib64/pkgconfig/atk.pc
```
now pkg-config seems to work
```
$ pkg-config --list-all | grep atk
atk                                 Atk - Accessibility Toolkit
pkg-config --libs atk.pc
-L/home/nfs/jack/.linuxbrew/Cellar/atk/2.28.1_2/lib64 -L/home/nfs/jack/.linuxbrew/Cellar/glib/2.56.1/lib -latk-1.0 -lgobject-2.0 -lglib-2.0
```
but gtk+ install keep failing
```
[jack@geo1 pkgconfig]$ brew install gtk+
Updating Homebrew...
==> Downloading https://download.gnome.org/sources/gtk+/2.24/gtk+-2.24.32.tar.xz
Already downloaded: /home/nfs/jack/.cache/Homebrew/gtk+-2.24.32.tar.xz
==> Downloading https://bug757187.bugzilla-attachments.gnome.org/attachment.cgi?id=331173
Already downloaded: /home/nfs/jack/.cache/Homebrew/gtk+--patch-ce5adf1a019ac7ed2a999efb65cfadeae50f5de8663638c7f765f8764aa7d931.cgi
==> Patching
==> Applying attachment.cgi
patching file gdk/quartz/gdkimage-quartz.c
==> ./configure --disable-silent-rules --prefix=/home/nfs/jack/.linuxbrew/Cellar/gtk+/2.24.32_2 --disable-glibtest --enable-introspection=yes --with-gdktarget=x11 --disable-visibility --disable
Last 15 lines from /home/nfs/jack/.cache/Homebrew/Logs/gtk+/01.configure:
checking for library containing strerror... none required
checking whether make sets $(MAKE)... (cached) yes
checking for pkg-config... /home/nfs/jack/.linuxbrew/opt/pkg-config/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for BASE_DEPENDENCIES... no
configure: error: Package requirements (glib-2.0 >= 2.28.0    atk >= 1.29.2    pango >= 1.20    cairo >= 1.6    gdk-pixbuf-2.0 >= 2.21.0) were not met:

No package 'atk' found
```

**Comment by iMichka** [comment]
Good to know that adding the symlink moved us a little bit forward. I think I will add some changes to brew and to the pkg-config formula to make this work. I’ll try to work on this today or tomorrow.

https://unix.stackexchange.com/questions/43190/where-did-usr-lib64-go-and-what-is-usr-lib-x86-64-linux-gnu

Reading this, it seems that we will need to support three types of folder layouts (lib, lib64, lib/x86-64-linux-gnu). And maybe more if needed.

I am still wondering who is deciding where to put stuff when atk is installed. Maybe it is make? As you all probably use your system’s make, this may explain the differences.

**Comment by sjackman** [comment]
Another solution is to move files from these 64-bit specific library directories to `lib`, since Linuxbrew does not support multilib.

**Comment by JuPlutonic** [comment]
I found problem with atk when using linuxbrew's python3.6.5 or ubuntu beaver's python3.6.5:

**/opt/.linuxbrew/bin/python3:**
Python 3.6.5 (default, Apr  3 2018, 15:18:52) 
[GCC 5.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gi
>>> gi.require_version("Gtk", "3.0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/.linuxbrew/Cellar/python/3.6.5/lib/python3.6/site-packages/gi/__init__.py", line 134, in require_version
    (namespace, version))
ValueError: Namespace Gtk not available for version 3.0
>>> from gi import Repository
>>> r = Repository.get_default()
>>> r.require('Gtk')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
gi.RepositoryError: Typelib file for namespace 'Atk', version '1.0' not found


**/usr/bin/python3:**
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gi
>>> gi.require_version("Gtk", "3.0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/gi/__init__.py", line 134, in require_version
    (namespace, version))
ValueError: Namespace Gtk not available for version 3.0
>>> from gi import Repository
>>> r = Repository.get_default()
>>> r.require('Gtk') # See note about versions
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
gi.RepositoryError: Typelib file for namespace 'Atk', version '1.0' not found
>>>

**Comment by JuPlutonic** [comment]
@arkanoid87  After I copied all from lib64 (x86_64-linux-gnu in my case): 
```
cp -R `brew --repository`/Cellar/atk/2.28.1_2/lib/x86_64-linux-gnu/* `brew --repository`/Cellar/atk/2.28.1_2/lib
```
got [another error (gist-file)](https://gist.github.com/a96954526320925b131f43d6fcd9ec40) so may be we need to fix atk \ at-sp2-atk to get **gtk+3** installation / bottling working.

**Comment by JuPlutonic** [comment]
After I copyed 
```
cp -R cp -R `brew --repository`/Cellar/at-spi2-core/2.28.0/lib/x86_64-linux-gnu/* cp -R `brew --repository`/Cellar/at-spi2-core/2.28.0/lib
```
I finally got **gtk+** installed.

**Comment by JuPlutonic** [comment]
@sjackman on your last comment.
Do we need to change Linuxbrew/homebrew-core/blob/master/Formula/pkg-config.rb and other env related tricks or change homebrew-core to install not to lib/lx86_64-linux-gnu?

**Comment by iMichka** [comment]
I found an (untested) solution: meson seems to accept a path for the libdir. We could modify the atk formula like this:
meson --libdir=pathtoinstallprerfix/lib

I need to test this but it may be a solution, as it will force the installation in the lib folder.

**Comment by iMichka** [comment]
Here is the a possible fix: #7663

Can somebody check if this works?

**Comment by JuPlutonic** [comment]
brew install --build-from-source /opt/.linuxbrew/Library/Taps/homebrew/homebrew-core/Formula/atk.rb
```
==> Downloading https://download.gnome.org/sources/atk/2.28/atk-2.28.1.tar.xz
Already downloaded: /home/salessionato/.cache/Homebrew/atk-2.28.1.tar.xz
==> Patching
patching file meson.build
==> meson --prefix=/opt/.linuxbrew/Cellar/atk/2.28.1_2 --libdir=/opt/.linuxbrew/
==> ninja
==> ninja install
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /opt/.linuxbrew
Could not symlink lib/pkgconfig/atk.pc
Target /opt/.linuxbrew/lib/pkgconfig/atk.pc
already exists. You may want to remove it:
  rm '/opt/.linuxbrew/lib/pkgconfig/atk.pc'

To force the link and overwrite all conflicting files:
  brew link --overwrite atk

To list all files that would be deleted:
  brew link --overwrite --dry-run atk

Possible conflicting files are:
/opt/.linuxbrew/lib/pkgconfig/atk.pc
==> Summary
/opt/.linuxbrew/Cellar/atk/2.28.1_2: 145 files, 3MB, built in 21 seconds
brew --overwrite --dry-run atk
Error: Unknown command: --overwrite
```
brew link --overwrite --dry-run atk
Would remove:
/opt/.linuxbrew/lib/pkgconfig/atk.pc

**Comment by sjackman** [comment]
@JuPlutonic Consider installing Linuxbrew in `/home/linuxbrew/.linuxbrew/` if possible so that you can use precompiled binary packages (known as bottles) for non-relocatable formula like `util-linux`.
Another possible workaround for you is `brew install --force-bottle util-linux`, but no promises.

If it's an option for you, you could open a ticket with your information systems department to ask that they create a `linuxbrew` role account with home directory `/home/linuxbrew`.

The precompiled binary bottles of non-relocatable bottles can only be used if you install in `/home/linuxbrew/.linuxbrew`, otherwise they have to be built from source. See the documentation below. On macOS the default installation directory is `/usr/local`. On Linux the default installation directory is `/home/linuxbrew/.linuxbrew`.

+ http://docs.brew.sh/Installation.html#alternative-installs
+ http://docs.brew.sh/FAQ.html#why-do-you-compile-everything
+ http://docs.brew.sh/Bottles.html#cellar-cellar

**Comment by iMichka** [comment]
@JuPlutonic good, you need to delete the symlink you manually created, and this should be fine.

This is now fixed by #7663
